src/azure-account.ts

```diff
@@ -783,71 +811,70 @@ async function getEnvironments(): Promise<AzureAccountEnvironment[]> {
-	const validateAuthority = activeDirectoryUrl.endsWith('/adfs') ? false : true;
-	if (apiProfile) {
-		return await getAzureStackEnvironments(ppe, validateAuthority)
-	} else {
-		return [
-			...staticEnvironments,
-			{
-				...ppe,
-				name: azurePPE,
-				validateAuthority: validateAuthority
-			}
-		]
+	const customCloudEnvironment: AzureAccountEnvironment | undefined = await getCustomCloudEnvironment(config, includePartial);
+	if (customCloudEnvironment) {
+		result.push(customCloudEnvironment);
 	}
+
+	return result;
 }
 
-async function getAzureStackEnvironments(ppe: Environment, validateAuthority: boolean) {
-	const resourceManagerUrl = ppe.resourceManagerEndpointUrl;
-	const endpointsUrl = getMetadataEndpoints(resourceManagerUrl);
-	const ppeResponse = await fetch(endpointsUrl);
-	if (ppeResponse.ok) {
-		const ppeMetadata: IResourceManagerMetadata = await ppeResponse.json();
-		return [
-			...staticEnvironments,
-			{
-				...ppe,
-				name: azurePPE,
-				portalUrl: ppeMetadata.portalEndpoint,
-				galleryEndpointUrl: ppeMetadata.galleryEndpoint,
-				activeDirectoryGraphResourceId: ppeMetadata.graphEndpoint,
-				storageEndpointSuffix: resourceManagerUrl.substring(resourceManagerUrl.indexOf('.')),
-				keyVaultDnsSuffix: '.vault'.concat(resourceManagerUrl.substring(resourceManagerUrl.indexOf('.'))),
-				managementEndpointUrl: ppeMetadata.authentication.audiences[0],
-				validateAuthority: validateAuthority,
-				azureStackApiProfile: true
-			}
-		]
-	} else {
-		return [
-			...staticEnvironments,
-			{
-				...ppe,
-				name: azurePPE,
-				validateAuthority: validateAuthority,
-				azureStackApiProfile: true
+async function getCustomCloudEnvironment(config: WorkspaceConfiguration, includePartial: boolean): Promise<AzureAccountEnvironment | undefined> {
+	const armUrl = config.get<string>(customCloudArmUrlKey);
+	if (armUrl) {
+		try {
+			const endpointsUrl = getMetadataEndpoints(armUrl);
+			const endpointsResponse = await fetch(endpointsUrl);
+			if (endpointsResponse.ok) {
+				const endpoints: IResourceManagerMetadata = await endpointsResponse.json();
+				return <AzureAccountEnvironment>{
+					name: azureCustomCloud,
+					resourceManagerEndpointUrl: armUrl,
+					activeDirectoryEndpointUrl: endpoints.authentication.loginEndpoint.endsWith('/') ? endpoints.authentication.loginEndpoint : endpoints.authentication.loginEndpoint.concat('/'),
+					activeDirectoryGraphResourceId: endpoints.graphEndpoint,
+					activeDirectoryResourceId: endpoints.authentication.audiences[0],
+					portalUrl: endpoints.portalEndpoint,
+					galleryEndpointUrl: endpoints.galleryEndpoint,
+					storageEndpointSuffix: armUrl.substring(armUrl.indexOf('.')),
+					keyVaultDnsSuffix: '.vault'.concat(armUrl.substring(armUrl.indexOf('.'))),
+					managementEndpointUrl: endpoints.authentication.audiences[0],
+					validateAuthority: getValidateAuthority(endpoints.authentication.loginEndpoint),
+					apiProfile: getApiProfile(),
```

Comment:
I thought @bganapa said we should ignore apiProfile for now.

I said [this](https://github.com/microsoft/vscode-azure-account/pull/251#discussion_r581293657):

> If we're trying to get this done faster, though, I think we could assume all CustomClouds are "2019-03-01-hybrid" for now and add support for the setting _later_ once we need to handle the other clouds.

He said this:

> Agree, a boolean may not be enough. lets go ahead as you proposed as we are trying to get this one faster

Reply:
If we _do_ want to do the setting, here are some things I think we need to figure out:

- I'm not sure the `apiProfile` setting fits well on `Environment`. Theoretically I could switch api profile regardless of cloud (Azure, Azure China, Custom, etc.) and at any time, which doesn't work well with the existing events like `onSessionsChanged`.
- We'd have to figure out how to tell people to switch _back_ to "latest" if they switch from a custom cloud to Azure global. Wouldn't want them stuck on the older profile.

We can figure those out, but it would be faster to just ignore apiProfile until we actually _need_ the setting

Reply:
@v-tsun as we discussed, lets remove the apiProfile setting and assume that if AzureCustomCloud, it is for AzureStack.

We can even go for explicit naming it as AzureStack instead of AzureCustomCloud? @ejizba

Reply:
I could go either way on the naming, but Storage Explorer uses "Custom", so I lean towards that
![Screen Shot 2021-02-24 at 11 49 55 AM](https://user-images.githubusercontent.com/11282622/109057346-7c72c380-7696-11eb-9057-d9cd7955c4d7.png)

Reply:
> I could go either way on the naming, but Storage Explorer uses "Custom", so I lean towards that
> ![Screen Shot 2021-02-24 at 11 49 55 AM](https://user-images.githubusercontent.com/11282622/109057346-7c72c380-7696-11eb-9057-d9cd7955c4d7.png)

Hey Eric, so you prefer to name it as AzureCustomCloud right? so @bganapa, how about we keep our name as AzureCustomCloud?